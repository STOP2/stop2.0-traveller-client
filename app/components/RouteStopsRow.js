import React, { Component } from 'react'
import { View } from 'react-native'
import { DefaultText } from '../components/textComponents'

import styles from '../styles/stylesheet'

class RouteStopsRow extends Component {
    constructor(props)
  {
        super(props)

        this.state = {}
    }

    render()
  {
        return (
      <View style={styles.busrow}>
        <DefaultText style={styles.busrowText}>{this.props.stopId}</DefaultText>
        <DefaultText style={styles.busrowTextBlack2}>{this.props.stopName}</DefaultText>
        <DefaultText style={styles.busrowTextBlack}>{this.props.arrivalTime} min</DefaultText>
      </View>)
    }
}

RouteStopsRow.propTypes = {
    stopId: React.PropTypes.string.isRequired,
    stopName: React.PropTypes.string.isRequired,
    arrivalTime: React.PropTypes.number.isRequired
}

export default RouteStopsRow
