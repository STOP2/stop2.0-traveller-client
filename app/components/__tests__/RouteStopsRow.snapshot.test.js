import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import RouteStopsRow from '../RouteStopsRow';

jest.mock('react-native-localization');

describe('RouteStopsRow component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<RouteStopsRow stopId="5124" stopName="Kumpulan kampus" arrivalTime={5} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
