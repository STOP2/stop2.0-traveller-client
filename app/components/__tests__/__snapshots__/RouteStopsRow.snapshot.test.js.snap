exports[`RouteStopsRow component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 60,
      "padding": 10,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "gotham-rounded-book",
        },
        Object {
          "flex": 1,
          "fontSize": 15,
        },
      ]
    }>
    5124
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "gotham-rounded-book",
        },
        Object {
          "color": "black",
          "flex": 3,
          "fontSize": 15,
        },
      ]
    }>
    Kumpulan kampus
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "gotham-rounded-book",
        },
        Object {
          "color": "black",
          "flex": 1,
          "fontSize": 15,
        },
      ]
    }>
    5
     min
  </Text>
</View>
`;
